ykutil/src/addr.rs

```diff
@@ -12,52 +12,58 @@ use std::{
 
 /// A Rust wrapper around `libc::Dl_info` using FFI types.
 ///
-/// The strings inside are handed out by the loader and can (for now, since we don't support
-/// dlclose) be considered of static lifetime. This makes the struct thread safe, and thus
-/// cacheable using `#[cached]`.
-#[derive(Debug, Clone)]
-pub struct DLInfo {
-    dli_fname: Option<&'static CStr>,
-    dli_fbase: usize,
-    dli_sname: Option<&'static CStr>,
-    dli_saddr: usize,
+/// The strings inside are handed out (as raw pointers) by the loader and can (for now, since we
+/// don't support dlclose) be considered of static lifetime (hence `unsafe impl` of `Send` and
+/// `Sync`). This makes the struct thread safe, and thus cacheable using `#[cached]`.
+/// #[derive(Debug)]
+pub struct DLInfo(libc::Dl_info);
+
+unsafe impl Send for DLInfo {}
+unsafe impl Sync for DLInfo {}
+
+/// `libc::DL_info` isn't `Clone`, so we have to jump some hoops.
```

Comment:
"jump THROUGH some"

Reply:
Fixed in 613ae16d2fe70052a3eb56fa8f065823d5e34dc1